File: multi-gammon-core/src/main/java/org/akoshterek/backgammon/agent/fa/AbsNeuralNetworkFA.java
package org.akoshterek.backgammon.agent.fa;

import org.encog.neural.networks.BasicNetwork;
import org.encog.persist.EncogDirectoryPersistence;

import java.io.File;

/**
 * @author Alex
 *         date 29.11.2015.
 */
public abstract class AbsNeuralNetworkFA implements NeuralNetworkFA {
    protected BasicNetwork network;

    public AbsNeuralNetworkFA(BasicNetwork network) {
        this.network = network;
    }

    public static BasicNetwork loadNN(File file) {
        return (BasicNetwork) EncogDirectoryPersistence.loadObject(file);
    }

    public static BasicNetwork loadNNFromResource(String resource) {
        return  (BasicNetwork)EncogDirectoryPersistence.loadResourceObject(resource);
    }
}

File: multi-gammon-core/src/main/java/org/akoshterek/backgammon/agent/fa/AbsNeuralNetworkFA.scala
package org.akoshterek.backgammon.agent.fa

import org.encog.neural.networks.BasicNetwork
import org.encog.persist.EncogDirectoryPersistence
import java.io.File

abstract class AbsNeuralNetworkFA(val network: BasicNetwork) extends NeuralNetworkFA {

}

object AbsNeuralNetworkFA {
    def loadNN(file: File): BasicNetwork = {
        return EncogDirectoryPersistence.loadObject(file).asInstanceOf[BasicNetwork]
    }

    def loadNNFromResource(resource: String): BasicNetwork = {
        return EncogDirectoryPersistence.loadResourceObject(resource).asInstanceOf[BasicNetwork]
    }
}

